### ISSUE TITLE
There is an issue where the mask does not apply correctly when using Mask Viewport Size and Adjust Scale simultaneously.

### ISSUE THREAD

**Issue by MizunagiKB** [MizunagiKB/gd_cubism#108]
The following is a quote from the original post: https://github.com/MizunagiKB/gd_cubism/issues/102#issuecomment-2301594475

> When I try to use Mask Viewport, it stops rendering the eyes (and other facial features such as blush), and I am not sure why.
> 
> UPD: "Adjust Scale" option is the culprit, if you have the scale less than 1, the mask viewport will be shifted towards (0;0) in scale. Mask Viewport does not follow the Scale properly.
> 
> 
> There is an issue where the mask does not apply correctly when using Mask Viewport Size and Adjust Scale simultaneously.

**Comment by MizunagiKB** [comment]
We have addressed an issue where the mask was not being referenced correctly when the canvas and mask sizes differed, and the default values for Scale or Adjust were changed.

The causes were as follows:

- When the mask size was small and scaling was applied, the scale could become less than one pixel, rendering the mask ineffective.
- When the canvas and mask sizes differed, the values specified for Adjust also needed to change proportionally, but they were not being recalculated correctly.

To address this issue, we made the following fixes:

- We ensured that scaling does not reduce the precision of the mask.
- Corresponding adjustments were made to the shader calculations.

This fix has been implemented in branch 0.7.
https://github.com/MizunagiKB/gd_cubism/tree/0.7

> Note: Branch 0.7 is still under development and has not been released yet. Additionally, it is intended for Godot 4.3 and above, so please apply with caution.
